lib/spack/spack/fetch_strategy.py

```diff
@@ -676,6 +681,9 @@ def __init__(self, **kwargs):
             raise ValueError(
                 "%s requires %s argument." % (self.__class__, self.url_attr))
 
+        if sys.platform == "win32":
+            self.url = self.url.replace('\\', '/')
```

Comment:
Absolutely agree, and noted on the issue board. I think this would be best handled along side a number of other Windows file path separator related issues in a separate PR.